grr/server/grr_response_server/flows/general/osquery_test.py

```diff
@@ -266,6 +271,130 @@ def testTotalRowCountIncludesAllChunks(self):
+
+    self.assertLen(results, 2)
+    self.assertEqual(type(results[0]), rdf_osquery.OsqueryResult)
+    self.assertEqual(type(results[1]), rdf_client_fs.StatEntry)
+
+    client_path = db.ClientPath.FromPathSpec(
+        self.client_id, results[1].pathspec)
+    byte_content = file_store.OpenFile(client_path).read()
+    self.assertEqual(len(byte_content), less_than_necessary_bytes)
+
+  def testFlowDoesntCollectSingleFileAboveTotalLimit(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      target_bytes = 2**20  # 1 MiB
+
+      phrase = "This is a > 1 MiB file\n"
+      phrase_bytes = len(phrase.encode("utf-8"))
+      repeat_times = target_bytes // phrase_bytes + 1
+      to_write = phrase.encode("utf-8") * repeat_times
+
+      less_than_necessary_bytes = target_bytes // 2
+
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write(to_write)
+
+      table = f"""
+      [
+        {{ "collect_collumn": "{temp_file_path}" }}
+      ]
+      """
+
+      with mock.patch.object(
+          osquery_flow,
+          "FILE_COLLECT_LIMIT_MAX_TOTAL_BYTES",
+          less_than_necessary_bytes):
+        with osquery_test_lib.FakeOsqueryiOutput(stdout=table, stderr=""):
+          results = self._RunFlow("Doesn't matter", ["collect_collumn"])
+
+    # Assert that only the OsqueryResult is returned (no file results)
+    self.assertLen(results, 1)
+    self.assertEqual(type(results[0]), rdf_osquery.OsqueryResult)
+
+  def testFlowCollectFile(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write("Just sample text to put in the file.".encode('utf-8'))
+
+      table = f"""
+      [
+        {{ "collect_column": "{temp_file_path}" }}
+      ]
+      """
+
+      with osquery_test_lib.FakeOsqueryiOutput(stdout=table, stderr=""):
+          results = self._RunFlow("Doesn't matter", ["collect_column"])
+      
+    self.assertLen(results, 2)
+    self.assertEqual(type(results[0]), rdf_osquery.OsqueryResult)
+    self.assertEqual(type(results[1]), rdf_client_fs.StatEntry)
+
+    pathspec = results[1].pathspec
+    client_path = db.ClientPath.FromPathSpec(self.client_id, pathspec)
+    fd_rel_db = file_store.OpenFile(client_path)
+    file_text = fd_rel_db.read().decode("utf-8")
+    self.assertEqual(file_text, "Just sample text to put in the file.")
+
+  def testFlowDoesntCollectWhenColumnsAboveLimit(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write("Just sample text to put in the file.".encode('utf-8'))
+
+      table = f"""
+      [
+        {{ "collect1": "{temp_file_path}", "collect2": "{temp_file_path}"}}
+      ]
+      """
+
+      with mock.patch.object(osquery_flow, "FILE_COLLECT_LIMIT_MAX_COLUMNS", 1):
+        with osquery_test_lib.FakeOsqueryiOutput(stdout=table, stderr=""):
+          with self.assertRaises(RuntimeError):
```

Comment:
Can we narrow down the exception?

Reply:
We will need to make changes to the test utils: https://github.com/google/grr/blob/v3.4.2.4-release/grr/test_lib/flow_test_lib.py#L426 . Do you reckon it is worth to do in this PR? We will also need to check if other tests use this `RuntimeError` I think.

Reply:
Ah, I see. Then in this case, RuntimeError seems sufficient.